src/cdk/schematics/ng-update/find-stylesheets.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {join} from '@angular-devkit/core';
+import {Tree} from '@angular-devkit/schematics';
+
+/** Regular expression that matches stylesheet paths */
+const STYLESHEET_REGEX = /.*\.(css|scss)/;
+
+/** Finds stylesheets in the given directory from within the specified tree. */
+export function findStylesheetFiles(tree: Tree, baseDir: string): string[] {
+  const result: string[] = [];
+  const visitDir = dirPath => {
+    // Do not visit directories or files inside node modules or `dist/` folders.
+    if (/^\/?(node_modules|dist)\//.test(dirPath)) {
+      return;
+    }
+    const stat = tree.getDir(dirPath);
+    stat.subdirs.forEach(fragment => visitDir(join(dirPath, fragment)));
+    result.push(...stat.subfiles.filter(f => STYLESHEET_REGEX.test(f)));
```

Comment:
nit: wouldn't be better if we pushed to the results first before visiting the sub-directories? That way the traversal should be in descending order.

Reply:
That shouldn't matter. I'll change it anyway as I agree that it makes more sense in descending order.